pkg/pillar/attest/attest.go

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package attest 
+
+import (
+	"fmt"
+	"errors"
+	"time"
+)
+
+type Event int
+type State int
+
+const (
+	EventInitialize = iota
+	EventNonceRecvd
+        EventInternalQuoteRecvd
+        EventRetryTimerExpiry
+        EventNonceMismatch
+        EventQuoteMismatch
+        EventNoQuoteCertRecvd
+        EventAttestEscrowRecorded
+	EventAttestEscrowFailed
+        EventAttestSuccessful
+	EventReAttest
+)
+
+const (
+	StateNone = iota
+        StateNonceWait
+        StateInternalQuoteWait
+        StateAttestWait
+        StateAttestEscrowWait
+	StateRestartWait
+        StateComplete
+)
+
+func eventToStr (event Event) string {
```

Comment:
Better to make these as member functions for the event type and call the function String().
That means that a %s print will do the right thing.